sql/src/main/java/org/apache/druid/sql/calcite/planner/CalciteRulesManager.java

```diff
@@ -82,6 +83,10 @@ public class CalciteRulesManager
   private static final int HEP_DEFAULT_MATCH_LIMIT = Integer.parseInt(
       System.getProperty(HEP_DEFAULT_MATCH_LIMIT_CONFIG_STRING, "1200")
   );
+  private static final String BLOAT_PROPERTY = "druid.sql.planner.bloat";
+  private static final int BLOAT = Integer.parseInt(
+      System.getProperty(BLOAT_PROPERTY, "100")
```

Comment:
I think the druid planner we'll not like at all if multiple projects are kep on top of eachother...

the [default](https://github.com/apache/calcite/blob/4e6a320bf476c4b3c313d86f8ff3c6edf4c2a578/core/src/main/java/org/apache/calcite/rel/rules/ProjectMergeRule.java#L49) seems to be the same (`100`) - what do you think about raising it to a bigger value; like a few millions - so that its less likely to hit this issue?

Reply:
Hi, considering that  this is primarily a workaround for a potential Calcite problem, I'm hesitant to increase the default value that much. However, I agree that raising the default to at least 1000 would be a good compromise.

Reply:
yeah - I agree...an off switch would be better than tweaking an integer;
it depends on the execution engine how it evaluates a set of expressions which may share common subexpressions ...it might not cause much problem in case it could avoid recomputations.